packages/lesswrong/server/resolvers/dialogueChecksResolvers.ts

```diff
@@ -0,0 +1,40 @@
+import DialogueChecks from "../../lib/collections/dialogueChecks/collection";
+import { randomId } from "../../lib/random";
+import { augmentFieldsDict } from "../../lib/utils/schemaUtils";
+import {defineMutation, defineQuery} from "../utils/serverGraphqlUtil";
+
+defineMutation({
+  name: "upsertUserDialogueCheck",
+  resultType: "DialogueCheck",
+  argTypes: "(targetUserId: String!, checked: Boolean!)",
+  fn: async (_, {targetUserId, checked}:{targetUserId:string, checked:boolean}, {currentUser, repos}) => {
+    if (!currentUser) throw new Error("No check user was provided")
+    if (!targetUserId) throw new Error("No target user was provided")
+    const existingCheck = await DialogueChecks.findOne({targetUserId, userId: currentUser._id})
```

Comment:
I am pretty sure you don't need this bit at all, given you're handling the upsert properly in `upsertDialogueCheck` with an `ON CONFLICT`.  (You'll generated a `randomId()` regardless of whether one already exists; it'll just only get used in the `DO UPDATE` case.)

Reply:
We need to return an id below for things to work. not addressing

Reply:
You don't need to do this, see follow-up in other thread